### PR TITLE
build EESSI compatibility layer v2023.06

### PR DESCRIPTION
**Pull request by trz42**
- duplicate PR for https://github.com/EESSI/compatibility-layer/pull/184
  - bumped version number
  - masked OpenSSL 3.*
- main reason we only want to build it for specific targets (so we need support for `bot: commands`)

### CODE REVIEW

**Comment by trz42** [comment]
bot: build inst:AWS arch:x86_64/generic repo:eessi-2023.04
bot: build inst:eX3 arch:aarch64/generic repo:eessi-2023.04

- received bot command `build inst:AWS arch:x86_64/generic repo:eessi-2023.04` from `trz42` (expanded command format: `build instance:AWS architecture:x86_64/generic repository:eessi-2023.04`)
- received bot command `build inst:eX3 arch:aarch64/generic repo:eessi-2023.04` from `trz42` (expanded command format: `build instance:eX3 architecture:aarch64/generic repository:eessi-2023.04`)

**Comment by trz42** [comment]
The above jobs `489226` and `4878` build for 2023.04 (not surprisingly). So new repo targets were added to the bot instances. See output of the command in subsequent PR comments

bot: showconfig
- received bot command `showconfig` from `trz42` (expanded command format: `showconfig `)

**Comment by trz42** [comment]
bot: build inst:AWS arch:x86_64/generic repo:eessi-2023.06-cl-bootstrap-prefix
bot: build inst:eX3 arch:aarch64/generic repo:eessi-2023.06-cl-build-node
- received bot command `build inst:AWS arch:x86_64/generic repo:eessi-2023.06-cl-bootstrap-prefix` from `trz42` (expanded command format: `build instance:AWS architecture:x86_64/generic repository:eessi-2023.06-cl-bootstrap-prefix`)
- received bot command `build inst:eX3 arch:aarch64/generic repo:eessi-2023.06-cl-build-node` from `trz42` (expanded command format: `build instance:eX3 architecture:aarch64/generic repository:eessi-2023.06-cl-build-node`)

**Comment by boegel** [comment]
@trz42 Don't we need to define a `2023.06` set in `EESSI/gentoo-overlay` first?

**Comment by trz42** [comment]
bot: build inst:AWS arch:x86_64/generic repo:eessi-2023.06-cl-bootstrap-prefix
bot: build inst:eX3 arch:aarch64/generic repo:eessi-2023.06-cl-build-node
- received bot command `build inst:AWS arch:x86_64/generic repo:eessi-2023.06-cl-bootstrap-prefix` from `trz42` (expanded command format: `build instance:AWS architecture:x86_64/generic repository:eessi-2023.06-cl-bootstrap-prefix`)
- received bot command `build inst:eX3 arch:aarch64/generic repo:eessi-2023.06-cl-build-node` from `trz42` (expanded command format: `build instance:eX3 architecture:aarch64/generic repository:eessi-2023.06-cl-build-node`)

File: ansible/playbooks/roles/compatibility_layer/defaults/main.yml [hunk 1]
```diff
@@ -26,6 +26,9 @@ prefix_mask_packages: |
   # stick to GCC 9.x; using a too recent compiler in the compat layer complicates stuff in the software layer,
   # see for example https://github.com/EESSI/software-layer/issues/151
   >=sys-devel/gcc-10
+  # see https://github.com/EESSI/software-layer/issues/257
+  # and https://github.com/EESSI/software-layer/issues/258
+  <dev-libs/openssl-3
```
Comment (boegel) [inline ansible/playbooks/roles/compatibility_layer/defaults/main.yml]: You want `>=` here, see also https://github.com/trz42/compatibility-layer/pull/2

Comment (trz42) [reply ansible/playbooks/roles/compatibility_layer/defaults/main.yml]: Thanks. Should be fixed now. Resubmitting jobs.

**Comment by trz42** [comment]
bot: build inst:AWS arch:x86_64/generic repo:eessi-2023.06-cl-bootstrap-prefix
bot: build inst:eX3 arch:aarch64/generic repo:eessi-2023.06-cl-build-node
- received bot command `build inst:AWS arch:x86_64/generic repo:eessi-2023.06-cl-bootstrap-prefix` from `trz42` (expanded command format: `build instance:AWS architecture:x86_64/generic repository:eessi-2023.06-cl-bootstrap-prefix`)
- received bot command `build inst:eX3 arch:aarch64/generic repo:eessi-2023.06-cl-build-node` from `trz42` (expanded command format: `build instance:eX3 architecture:aarch64/generic repository:eessi-2023.06-cl-build-node`)

**Comment by trz42** [comment]
The jobs `4882` and `489409` likely failed because some package needed for direnv could not be downloaded. That's fixed now. Resubmitting jobs.

bot: build inst:AWS arch:x86_64/generic repo:eessi-2023.06-cl-bootstrap-prefix
bot: build inst:eX3 arch:aarch64/generic repo:eessi-2023.06-cl-build-node
- received bot command `build inst:AWS arch:x86_64/generic repo:eessi-2023.06-cl-bootstrap-prefix` from `trz42` (expanded command format: `build instance:AWS architecture:x86_64/generic repository:eessi-2023.06-cl-bootstrap-prefix`)
- received bot command `build inst:eX3 arch:aarch64/generic repo:eessi-2023.06-cl-build-node` from `trz42` (expanded command format: `build instance:eX3 architecture:aarch64/generic repository:eessi-2023.06-cl-build-node`)

**Comment by trz42** [comment]
Repeating jobs after interactively run builds seem to work.

bot: build inst:AWS arch:x86_64/generic repo:eessi-2023.06-cl-bootstrap-prefix
bot: build inst:eX3 arch:aarch64/generic repo:eessi-2023.06-cl-build-node
- received bot command `build inst:AWS arch:x86_64/generic repo:eessi-2023.06-cl-bootstrap-prefix` from `trz42` (expanded command format: `build instance:AWS architecture:x86_64/generic repository:eessi-2023.06-cl-bootstrap-prefix`)
- received bot command `build inst:eX3 arch:aarch64/generic repo:eessi-2023.06-cl-build-node` from `trz42` (expanded command format: `build instance:eX3 architecture:aarch64/generic repository:eessi-2023.06-cl-build-node`)

**Comment by trz42** [comment]
bot: build inst:AWS arch:aarch64/generic repo:eessi-2023.06-cl-bootstrap-prefix
bot: build inst:eX3 arch:aarch64/generic repo:eessi-2023.06-cl-bootstrap-prefix
- received bot command `build inst:AWS arch:aarch64/generic repo:eessi-2023.06-cl-bootstrap-prefix` from `trz42` (expanded command format: `build instance:AWS architecture:aarch64/generic repository:eessi-2023.06-cl-bootstrap-prefix`)
- received bot command `build inst:eX3 arch:aarch64/generic repo:eessi-2023.06-cl-bootstrap-prefix` from `trz42` (expanded command format: `build instance:eX3 architecture:aarch64/generic repository:eessi-2023.06-cl-bootstrap-prefix`)

**Pull request merged**